src/internal/storage/track/renewer.go

```diff
@@ -67,7 +74,13 @@ func (r *Renewer) Close() (retErr error) {
 		}
 	}()
 	ctx := context.Background()
-	_, err := r.tracker.SetTTLPrefix(ctx, r.id+"/", ExpireNow)
+	_, n, err := r.tracker.SetTTLPrefix(ctx, r.id+"/", ExpireNow)
+	if err != nil {
+		return err
+	}
+	if n != r.n {
```

Comment:
We should add a TODO to make it such that we can fail earlier when this happens. A long upload will fail right at the end when this issue is hit.